internal/events/event.go

```diff
@@ -261,64 +357,123 @@ func (e Events) queryEvents(ctx context.Context, selectedCategories []string, cl
-	db := e.db.Order("event_time").Where("category IN (?)", selectedCategories)
-	if e.authz != nil {
-		db = e.authz.OwnedBy(ctx, db)
-	}
-
-	var result *gorm.DB
-
 	//retrieveing all events can be done only by admins. This is done to restrict data
 	//intensive queries by common users
 	if allEvents() && e.authz.IsAdmin(ctx) {
-		result = db
+		return tx
 	}
 
 	//for bound events that are searched with cluster id (whether on clusters, bound infra-env ,
 	//host bound to a cluster or registered to a bound infra-env) check the access permission
 	//relative to the cluster ownership
 	if clusterBoundEvents() {
-		result = db.Model(&common.Cluster{}).Select("events.*, clusters.user_name, clusters.org_id").
-			Joins("INNER JOIN \"events\" ON events.cluster_id = clusters.id")
+		tx = tx.Model(&common.Cluster{}).
+			Select("events.*, clusters.user_name, clusters.org_id").
+			Joins("INNER JOIN events ON clusters.id = events.cluster_id")
+
+		// if deleted hosts flag is true, we need to add 'deleted_at' to know whether events are related to a deleted host
+		if swag.BoolValue(deletedHosts) {
+			tx = tx.Select("events.*, clusters.user_name, clusters.org_id, hosts.deleted_at").
+				Joins("LEFT JOIN hosts ON events.host_id = hosts.id")
+		}
+		return tx
 	}
 
 	//for unbound events that are searched with infra-env id (whether events on hosts or the
 	//infra-env level itself) check the access permission relative to the infra-env ownership
 	if nonBoundEvents() {
-		result = db.Model(&common.InfraEnv{}).Select("events.*, infra_envs.user_name, infra_envs.org_id").
-			Joins("INNER JOIN events ON events.infra_env_id = infra_envs.id")
+		return tx.Model(&common.InfraEnv{}).
+			Select("events.*, infra_envs.user_name, infra_envs.org_id").
+			Joins("INNER JOIN events ON infra_envs.id = events.infra_env_id")
 	}
 
 	//for query made on the host only check the permission relative to it's infra-env. since
 	//host table does not contain an org_id we can not perform a join on that table and has to go
 	//through the infra-env table which is good because authorization is done on the infra-level
 	if hostOnlyEvents() {
-		result = db.Model(&common.Host{}).Select("events.*, infra_envs.user_name, infra_envs.org_id").
+		return tx.Model(&common.Host{}).Select("events.*, infra_envs.user_name, infra_envs.org_id").
 			Joins("INNER JOIN infra_envs ON hosts.infra_env_id = infra_envs.id").Joins("INNER JOIN events ON events.host_id = hosts.id")
 	}
 
-	if result == nil { //non supported option
-		return make([]*common.Event, 0), nil
+	//non supported option
+	return nil
+}
+
+func preparePaginationParams(limit, offset *int64) (*int64, *int64) {
+	if limit == nil {
+		// If limit is not valid (smaller than 1 or equals 0), we set it -1 (no limit).
+		limit = swag.Int64(-1)
+	} else if *limit < -1 || *limit == 0 {
+		// If limit not valid (smaller than 1 or equals 0), we set it a default (currently 5000).
+		limit = swag.Int64(defaultEventLimit)
+	}
```

Comment:
it's a bit of a confusing behavior. I would do:
* if ``limit`` is null (not provided), let it use the default
* if ``limit`` is less than zero, be infinite 
* if someone wants ``limit=0``, that's ok. the code should return an empty list by itself